### PR TITLE
Adding RxActivityResult

### PR DESCRIPTION
**Pull request by manimaul**
This PR adds the ability to [startActivityForResult()](https://developer.android.com/reference/android/app/Activity.html#startActivityForResult) via a simple Rx API.  Under the hood it uses a "headless behavioral Fragment" to broker the transaction and transmit the resulting event. Note that callers no longer need to concern themselves with "request codes" or implementing callbacks.

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -21,6 +21,7 @@ ext {
   supportTestRules = 'com.android.support.test:rules:0.5'
   supportTestEspresso = 'com.android.support.test.espresso:espresso-core:2.2.1'
   supportTestEspressoContrib = 'com.android.support.test.espresso:espresso-contrib:2.2.1'
+  rxRelay = 'com.jakewharton.rxrelay:rxrelay:1.2.0'
```
Comment (manimaul) [inline dependencies.gradle]: Rx Relay is used only in a test.

**Comment by manimaul** [comment]
The generateKotlin task generated invalid Kotlin.  Would you be interested in updating the task to skip generation if Kotlin has been manually committed... or ?

**Comment by manimaul** [comment]
Closing PR - I'll re-open after working on familiarizing with the project build pipeline and tests.

**Comment by manimaul** [comment]
It looks like the tests failed because of an un-related failure of RxSwipeDismissBehaviorTest.  There is a similar failure on build [#754](https://travis-ci.org/JakeWharton/RxBinding/builds/173980154) the main branch.

**Comment by JakeWharton** [comment]
Thanks for taking the time to send a PR, but unfortunately this isn't something I'm comfortable including in the library. RxBindings focuses on the 'view' and 'widget' packages only as a way to scope itself reasonably and not attempt to provide observables for the entire framework. Since this deals with activities who themselves have lots of interesting things that could be observed it doesn't fall within the scope I've defined.

Sorry I didn't reply sooner but I've been busy with other projects and neglected this one a bit.

**Comment by manimaul** [comment]
Perhaps you should consider adding module to your repo for non-view widget "RxBinding".  If that is something you would be interested in then I'd be willing to work on it.  There is a ton of value in simplifying and "Rx-ifying" some of Android APIs such as requesting permissions and starting and activity for a result etc.

**Comment by JakeWharton** [comment]
There are a lot of separate projects for other bindings. Things like
permissions, broadcasts, preferences, location, bluetooth, etc. There's far
too many to maintain in a single repo and it would only complicate this
project (which is already quite complicated!) even further.

**Pull request closed without merging**